gce/survival-training/README.md

```diff
@@ -0,0 +1,93 @@
+# GCE survival training
+
+Training jobs that survive GCE shutdowns
+
+- - -
+
+[Google Compute Engine](https://cloud.google.com/compute/docs/) (GCE) gives you direct access to the computational backbone of Google Cloud Platform. You can use GCE to create virtual machines provisioned with the resources of your choosing ([satisfying the constraints of any quotas that you define](https://cloud.google.com/compute/quotas)).
+
+This has particular value when it comes to training machine learning models, as this gives you access to resources beyond those available to you physically. For example, GCE will soon offer preemptible GPU instances. This means that, if you designed your trainer appropriately, your cost per training step could be a fraction of what it would be on a dedicated instance.
+
+Even if you want to train on a dedicated instance, GPU-enabled GCE instances have always had the issue that they go down about once a week for maintenance. This introduces a significant amount of overhead (in both time and money) to performing training on GCE instances.
+
+However, GCE *does* expose primitives that allow you set up training jobs and your GCE instances in a manner that lets you train across instance preemptions and shutdowns.
+
+This guide provides you with a template to do exactly that.
+
+
+## Survival strategy
+
+There are five components to our resilient training jobs:
+
+1. [GCE custom images](https://cloud.google.com/compute/docs/images#custom_images)
+
+1. A trainer command line interface
+
+1. [GCE startup scripts](https://cloud.google.com/compute/docs/startupscript)
+
+1. [GCE instance metadata](https://cloud.google.com/compute/docs/storing-retrieving-metadata)
+
+1. [Cloud Source Repositories](https://cloud.google.com/source-repositories/)
+
+
+As a part of this process, if you have not already done so, you will be prompted to enable both the Compute Engine API as well as the Cloud Source Repositories API for your Google Cloud Platform project.
```

Comment:
probably make this (and the next line) into something like "before you start" kind of steps.